pluggy/hooks.py

```diff
@@ -233,6 +233,9 @@ def remove(wrappers):
             if remove(self._nonwrappers) is None:
                 raise ValueError("plugin %r not found" % (plugin,))
 
+    def get_hookimpl(self):
```

Comment:
The semantics are bit off since there is of course a collection of callback functions.
I would go with `get_hookimpls()`.

Also since in `pluggy.manager` it seems every use of `_wrappers` and `_nonwrappers` is a concatenation,
```python
 >>> git grep _nonwrappers                                                                        (git)-[master]
pluggy/hooks.py:        self._nonwrappers = []
pluggy/hooks.py:            if remove(self._nonwrappers) is None:
pluggy/hooks.py:            methods = self._nonwrappers
pluggy/hooks.py:        return self._hookexec(self, self._nonwrappers + self._wrappers, kwargs)
pluggy/hooks.py:        res = self._hookexec(self, self._nonwrappers + self._wrappers, kwargs)
pluggy/hooks.py:        old = list(self._nonwrappers), list(self._wrappers)
pluggy/hooks.py:            self._nonwrappers, self._wrappers = old
pluggy/manager.py:                    for hookfunction in hc._wrappers + hc._nonwrappers:
pluggy/manager.py:                    for hookimpl in hook._wrappers + hook._nonwrappers:
pluggy/manager.py:            for hookimpl in orig._wrappers + orig._nonwrappers:
```
 I'd just convert all lines containing `hc._wrappers + hc._nonwrappers` to use this call *now*.

Reply:
Agree with both points, seems harmless as the PR is quite small as is.

Reply:
So currently the `get_hookimpls()` is always returning a list with both wrappers and nonwrappers.

To replace, for instance, `old = list(self._nonwrappers), list(self._wrappers)`, then we would have to possibly add an option to return them separately.

I cannot think of a very good way of doing it, so for now at least I won't touch this specific part. And it is not that bad as it is accessing its own private variable at least.

Reply:
Also, `self._hookexec(self, self._nonwrappers + self._wrappers, kwargs)` turns out to be strictly dependent on this order "nonwrappers then wrappers". I changed `get_hookimpls()` to be like this, hoping there will be no side-effects in the other ones that had the original order (tests are passing it at least).